sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/SynchronousReceiveWork.java

```diff
@@ -43,6 +57,20 @@
         this.numberToReceive = numberToReceive;
         this.timeout = timeout;
         this.emitter = emitter;
+
+        nextMessageSubscriber = Flux.switchOnNext(messageReceivedEmitter
+            .map(messageContext -> {
+                emitter.next(messageContext);
```

Comment:
You don't need any logic to be in here. In the onNext() override, is where you would call emitter.next. And wherever the flux is attached to the Emitter could be the messageReceivedEmitter.